### DISCUSSION TITLE
Piano keyboard

### DISCUSSION THREAD

**Discussion by stosem** [SynthstromAudible/DelugeFirmware#623] [Ideas]
In addition to the isomorphic keyboard, with a 16x8 button matrix, we can create an 8 octave piano keyboard.
And bind this piano keyboard to the second click of the "Keyboard" button (or switch keyboard type in the Settings menu).

**Comment by litui** [comment]
Is this a feature suggestion or a commitment to work on this item?  Feature suggestions generally go in the Synthstrom Forum.

**Comment by stosem** [comment]
Yes, I would like to work on it. But not sure about my time and possibly, so feel free to close this ticket as a feature request. Sorry. I can reopen it if I (or someone else) will be ready to work.
Thanks!

**Comment by bfredl** [comment]
Would this be addressed by using #46 and setting the row spacing to 12 semitones?

**Comment by stosem** [comment]
It may looks like this. So, dynamic population is possible, using loops and intervals, but seems to have an additional logic. ![16868181319144260124609205097051](https://github.com/SynthstromAudible/DelugeFirmware/assets/21126513/fe32a576-b391-4dd2-a75a-9d2db8289ca3)

**Comment by PaulFreund** [comment]
@stosem Do you still want to work on this? I'm planning to start work on a PR with a multi layout (piano, isomorphic, fixed), configurable (rows, columns, in-key only, chords) keyboard that should implement #55 and #19 extending on #46 and don't want to collide with anybody

**Comment by stosem** [comment]
> I'm planning to start work on a PR with a multi layout 

That would be nice! Yes, start please, I'll check you MR and maybe join your work with testing or any other help. Thanks!

**Comment by PaulFreund** [comment]
Hi @stosem, the Keyboard Layout API is done and merged into community. Maybe give it a look and tell me if you would like to implement it or if I should do it :)

**Comment by sichtbeton** [comment]
this could be close IMHO. either make a PR out of it or close it. its not an issue but rather a feature suggestion.
